app/components/istag-select/istag-select.component.js

```diff
@@ -0,0 +1,158 @@
+
+    var ctrl = this;
+    var imageStreamsVersion = APIService.getPreferredVersion('imagestreams');
+    ctrl.isByNamespace = {};
+    ctrl.isNamesByNamespace = {};
+
+    // Check if the istag object contains data about namespace/imageStream/tag so the ui-select will be pre-populated with them
+    var shouldPrepopulate = _.get(ctrl, 'istag.namespace') && _.get(ctrl, 'istag.imageStream') && _.get(ctrl, 'istag.tagObject.tag');
+
+    var ensureStatusTags = function(imageStreams) {
+      // Make sure each image stream has a status tags array, even if empty.
+      _.each(imageStreams, function(imageStream) {
+        if (!_.get(imageStream, 'status.tags')) {
+          _.set(imageStream, 'status.tags', []);
+        }
+      });
+    };
+
+    var prepopulate = function(ns) {
+      ctrl.isByNamespace[ns] = {};
+      ctrl.isNamesByNamespace[ns] = [];
+
+      if (!_.includes(ctrl.namespaces, ns)) {
+        ctrl.namespaces.push(ns);
+        ctrl.isNamesByNamespace[ns] = ctrl.isNamesByNamespace[ns].concat(ctrl.istag.imageStream);
+        ctrl.isByNamespace[ns][ctrl.istag.imageStream] = {status: {tags: [{tag: ctrl.istag.tagObject.tag}]}};
+        return;
+      }
+      DataService.list(imageStreamsVersion, { namespace: ns }, function(isData) {
+        // Make a copy since we modify status tags and don't want to mutate objects that are cached.
+        var imageStreams = angular.copy(isData.by('metadata.name'));
+        ensureStatusTags(imageStreams);
+        ctrl.isByNamespace[ns] = imageStreams;
+        ctrl.isNamesByNamespace[ns] = _.keys(imageStreams).sort();
+
+        //  Image stream is missing
+        if (!_.includes(ctrl.isNamesByNamespace[ns], ctrl.istag.imageStream)) {
+          ctrl.isNamesByNamespace[ns] = ctrl.isNamesByNamespace[ns].concat(ctrl.istag.imageStream);
+          ctrl.isByNamespace[ns][ctrl.istag.imageStream] = {
+            status: {
+              tags: {}
+            }
+          };
+        }
+
+        // Tag is missing
+        if (!_.find( ctrl.isByNamespace[ns][ctrl.istag.imageStream].status.tags, {tag: ctrl.istag.tagObject.tag})) {
+          ctrl.isByNamespace[ns][ctrl.istag.imageStream].status.tags.push({tag: ctrl.istag.tagObject.tag});
+        }
+      });
+    };
+
+    ProjectsService.list().then(function(projectData) {
+      ctrl.namespaces = _.keys(projectData.by('metadata.name'));
+
+      if (ctrl.includeSharedNamespace) {
+        // Use _.uniq to avoid adding "openshift" twice if the user is a
+        // member of the openshift namespace.
+        ctrl.namespaces = _.uniq(['openshift'].concat(ctrl.namespaces));
+      }
+
+      ctrl.namespaces = ctrl.namespaces.sort();
+      ctrl.namespaceChanged(ctrl.istag.namespace);
+    });
+
+    ctrl.namespaceChanged = function (namespace){
+      // In case the selectboxes dont have to be prepopulated, which means the namespace selectbox
+      // value was changed by user, clear the imageStream and tag.
+      if (!shouldPrepopulate) {
```

Comment:
Are we sure this is necessary? According to the ng-change doc, it should only fire if the user changes the value. But maybe ui-select isn't following that.

https://code.angularjs.org/1.5.11/docs/api/ng/directive/ngChange

Reply:
yes, cause we are calling the `namespaceChanged()` method to initialize the imageStream and tag selectboxes on line ~111.
Ultimately we could drop the `shouldPrepopulate` var and add an extra bool argument that would handle it.